doc/high-level.md

```diff
@@ -0,0 +1,83 @@
+# High Level Bindings
+
+The purpose of the high-level magicl bindings is to allow for MATLAB-like multidimensional arrays in lisp. 
+
+## Constructors
+
+The construction of tensors can be done with any of the given constructors. The constructors take a shape and the arguments for method of construction.
+
+Tensors are specialized on both the shape and the element type. The class of a tensor will be of the form `$CLASS/$TYPE`. (eg. `MATRIX/DOUBLE-FLOAT`)
+
+| Number of dimensions | Tensor Class    |
+|----------------------|-----------------|
+| 1                    | `VECTOR`        |
+| 2                    | `MATRIX`        |
+| *                    | `TENSOR`        |
+
+| Element Type             | Class Suffix           |
+|--------------------------|------------------------|
+| `SINGLE-FLOAT`           | `SINGLE-FLOAT`         |
+| `DOUBLE-FLOAT`           | `DOUBLE-FLOAT`         |
+| `(COMPLEX SINGLE-FLOAT)` | `COMPLEX-SINGLE-FLOAT` |
+| `(COMPLEX DOUBLE-FLOAT)` | `COMPLEX-DOUBLE-FLOAT` |
+| `(SIGNED-BYTE 32)`       | `INT32`                |
+
+The type of the elements of the tensor can be specified with the `:type` keyword, or the constructor will attempt to find an appropriate type from the given arguments. The default element type for a tensor is `DOUBLE-FLOAT`.
+
+The layout of the tensor (column-major or row-major) can be specified with the `:layout` keyword. This affects the underlying storage of the tensor and will affect how it carries out operations with LAPACK. 
+
+
+## Other Library Equivalents
+
+This table was adapted largely from the [NumPy Equivalent Table](https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html#linear-algebra-equivalents).
+
+### Basic Accessors
+
+| MAGICL         | MATLAB     | NumPy                   | Description                                                   |
+|----------------|------------|-------------------------|---------------------------------------------------------------|
+| `(layout a)`    | `ndims(a)` | `ndim(a)` or `a.ndim`   | Get the number of dimensions of the array.                    |
```

Comment:
`layout` is presently undefined for vectors. For matrices and tensors, it seems to return something like `:column-major` rather than the number of dimensions.

Reply:
Fixed 400a40c